File: python/src/ev1/ut1_fundamentospython/u02tiposvariables/teoria/U04_OperadoresComparativos.py
"""
U04 — Operadores de comparacion.

Los comparadores son como los booleanos, devuelen "True"
si algo es cierto o verdadero y devuelven "False" si algo
es falso o no es cierto.
OBJETIVO:
  - Comparar con == != < > <= >=

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
"""
print("\nINICIO DEL PROGRAMA\n")

ingresos_mes_anterior = 1400.56
ingresos_mes_actual = 2345.12
valor_pi_ayer = 3.1416
valor_pi_hoy = 3.1416

# COMPARADOR DE IGUALDAD: El operador usado para verificar si dos datos son iguales es el doble signo de igual ("==").
print("\¡PRUEBA COMPARANDO LA IGUALDAD DE VALORES! (==)\n")

respuesta = ingresos_mes_actual == ingresos_mes_anterior
print(f"¿Son iguales los ingresos de este mes a los del mes anterior? La respuestas es: {respuesta}. \n")

respuesta = valor_pi_ayer == valor_pi_hoy
print(f"¿El valor de PI de ayer es el mismo de hoy? La respuesta es: {respuesta}. \n")

# COMPARADOR DE DESIGUALDAD: El operador usado para verificar si dos datos son diferentes es el signo de exclamaciín con un signo de igual ("!=").
print("\¡PRUEBA COMPARANDO LA DIFERENCIA DE VALORES! (!=)\n")
respuesta = ingresos_mes_actual != ingresos_mes_anterior
print(f"¿Los ingresos del mes anterior y del nes actual son diferentes? Respuesta: {respuesta}. \n")

print(f"¿El valor de PI de ayer es diferente al valor de PI de hoy? Respuesta: {valor_pi_ayer != valor_pi_hoy}. \n")

# COMPARADOR DE MAYOR: El operador usado para verificar si un "X" dato es mayor que otro es el signo de mayor (">").
print("\¡PRUEBA COMPARANDO CUAL DE LOS VALORES ES MAYOR! (>)\n")

print(f"¿Los ingresos de este mes superaron los del mes anterior? La respuesta es {ingresos_mes_actual > ingresos_mes_anterior}. \n")
print(f"¿El valor de PI de ayer es mayor al valor de PI de hoy? La respuesta es: {valor_pi_ayer > valor_pi_hoy}. \n")

# COMPARADOR DE MENOR: El operador usado para verificar si un "X" dato es menor que otro es el signo de mayor ("<").
print("\¡PRUEBA COMPARANDO CUAL DE LOS VALORES ES MENOR! (<)\n")

print(f"¿Los ingresos del mes anterior superaron los del mes actual? La respuesta es {ingresos_mes_actual < ingresos_mes_anterior}. \n")
print(f"¿El valor de PI de ayer es menor que el valor de PI de hoy? La respuesta es: {valor_pi_ayer < valor_pi_hoy}. \n")

# COMPARADOR DE MAYOR O IGUAL: El operador usado para verificar si un "X" dato es mayor o igual que otro es el signo de mayor y el signo de igual. (">=").

# TODO: Falta explicar y terminar de hacer esto.

# COMPARADOR DE MAYOR O IGUAL: El operador usado para verificar si un "X" dato es menor o igual que otro es el signo de menor y el signo de igual. ("<=").

# TODO: Falta explicar y terminar de hacer esto.

print("\nFIN DEL PROGRAMA\n")

File: csharp/src/ev1/ut1_fundamentoscsharp/u02tiposvariables/teoria/U04_OperadoresComparativos.cs
/*
U04 — Operadores de comparacion.

Los comparadores devuelven bool: true si la relacion se cumple, false si no.

OBJETIVO:
  - Usar == != < > <= >=
  - Guardar el resultado en un bool o imprimirlo directamente.
  - Comparar numeros y tambien strings (igualdad de contenido).

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

Console.WriteLine("\n¡INICIO DEL PROGRAMA!\n");

double ingresosMesAnterior = 1400.56;
double ingresosMesActual = 2345.12;
double valorPiAyer = 3.1416;
double valorPiHoy = 3.1416;

// == igualdad
Console.WriteLine("¡PRUEBA IGUALDAD (==)!\n");
bool respuesta = ingresosMesActual == ingresosMesAnterior;
Console.WriteLine($"¿Ingresos mes actual == mes anterior? {respuesta}");
respuesta = valorPiAyer == valorPiHoy;
Console.WriteLine($"¿PI de ayer == PI de hoy? {respuesta}\n");

// != desigualdad
Console.WriteLine("¡PRUEBA DESIGUALDAD (!=)!\n");
Console.WriteLine($"¿Ingresos distintos? {ingresosMesActual != ingresosMesAnterior}");
Console.WriteLine($"¿PI distinto? {valorPiAyer != valorPiHoy}\n");

// >
Console.WriteLine("¡PRUEBA MAYOR (>)!\n");
Console.WriteLine($"¿Ingresos actuales > anteriores? {ingresosMesActual > ingresosMesAnterior}");
Console.WriteLine($"¿PI ayer > PI hoy? {valorPiAyer > valorPiHoy}\n");

// <
Console.WriteLine("¡PRUEBA MENOR (<)!\n");
Console.WriteLine($"¿Ingresos actuales < anteriores? {ingresosMesActual < ingresosMesAnterior}");
Console.WriteLine($"¿PI ayer < PI hoy? {valorPiAyer < valorPiHoy}\n");

// >= y <=
Console.WriteLine("¡PRUEBA MAYOR O IGUAL (>=) Y MENOR O IGUAL (<=)!\n");
int nota = 5;
Console.WriteLine($"nota ({nota}) >= 5 ? {nota >= 5}");
Console.WriteLine($"nota ({nota}) <= 4 ? {nota <= 4}");
Console.WriteLine($"nota ({nota}) >= 10 ? {nota >= 10}\n");

// Comparacion de strings
Console.WriteLine("¡COMPARAR STRINGS!\n");
string ciudadA = "Sevilla";
string ciudadB = "Sevilla";
string ciudadC = "Madrid";
Console.WriteLine($"\"{ciudadA}\" == \"{ciudadB}\" ? {ciudadA == ciudadB}");
Console.WriteLine($"\"{ciudadA}\" == \"{ciudadC}\" ? {ciudadA == ciudadC}");

Console.WriteLine("\n¡FIN DEL PROGRAMA!\n");
